pkgs/servers/x11/xorg/update.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env nix-shell
+#!nix-shell --pure --keep NIX_PATH -i python3 -p nix git python3Packages.packaging python3Packages.beautifulsoup4 python3Packages.requests
+
+# Usage: Run ./update.py from the directory containing tarballs.list. The script checks for the
+# latest versions of all packages, updates the expressions if any update is found, and commits
+# any changes.
+
+import subprocess
+
+import requests
+from bs4 import BeautifulSoup
+from packaging import version
+
+mirror = "mirror://xorg/"
+allversions = {}
+
+print("Downloading latest version info...")
+
+# xorg packages
+for component in [
+    "individual/app",
+    "individual/data",
+    "individual/data/xkeyboard-config",
+    "individual/doc",
+    "individual/driver",
+    "individual/font",
+    "individual/lib",
+    "individual/proto",
+    "individual/util",
+    "individual/xcb",
+    "individual/xserver",
+]:
+    url = "https://xorg.freedesktop.org/releases/{}/".format(component)
+    r = requests.get(url)
+    soup = BeautifulSoup(r.text, "html.parser")
+    for a in soup.table.find_all("a"):
+        href = a["href"]
+        if (
+            not href.endswith(".tar.bz2")
+            and not href.endswith(".tar.gz")
+            and not href.endswith(".tar.xz")
+        ):
+            continue
+
+        pname, rem = href.rsplit("-", 1)
+        ver, _, ext = rem.rsplit(".", 2)
+
+        if "rc" in ver:
+            continue
+
+        entry = allversions.setdefault(f"{mirror}{component}/{pname}", ([], {}))
+        entry[0].append(version.parse(ver))
+        entry[1][ver] = f"{mirror}{component}/{href}"
+
+# luit
+lurl = "https://invisible-mirror.net/archives/luit/"
+r = requests.get(lurl)
+soup = BeautifulSoup(r.text, "html.parser")
+for a in soup.find_all("a"):
+    href = a["href"]
+    if href.endswith(".tgz"):
+        pname, rem = href.rsplit("-", 1)
+        ver, _ = rem.rsplit(".", 1)
+
+        entry = allversions.setdefault(f"{lurl}{pname}", ([], {}))
+        entry[0].append(version.parse(ver))
+        entry[1][ver] = f"{lurl}{href}"
+
+print("Finding updated versions...")
+
+with open("./tarballs.list") as f:
+    lines_tarballs = f.readlines()
+
+updated_tarballs = []
+changes = {}
+changes_text = []
+for line in lines_tarballs:
+    line = line.rstrip("\n")
+    if any(line.startswith(frag) for frag in [mirror, lurl]):
```

Comment:
I think we can use `continue` here:
```python
    if not line.startswith((mirror, lurl)):
        continue
    pname, rem = line.rsplit("-", 1)
    ...
```

Reply:
This was the bug. Lines should be always added to tarballs file, with or without changes.